Update template's usage of Entity Framework tools

natemcmaster opened the pull request:
Cref https://github.com/aspnet/EntityFramework/pull/5635. This is in for preview 2.

For preview 2 and beyond, we will be instructing users to put "Microsoft.EntityFrameworkCore.Tools" in the "tools" section of project.json, and "Microsoft.EntityFrameworkCore.Design" into "dependencies".

cc @phenning

dnfclas commented:
Hi **@natemcmaster**, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

phenning commented on src/Rules/StarterWeb/AI/IndividualAuth/project.json:
```diff
@@ -20,7 +20,8 @@
     "Microsoft.EntityFrameworkCore.SqlServer": "1.0.0-*",
-    "Microsoft.EntityFrameworkCore.Tools": {
+    "Microsoft.EntityFrameworkCore.SqlServer.Design": "1.0.0-*",
+    "Microsoft.EntityFrameworkCore.Design": {
```
Alphabetic order, or build will fail

phenning commented on src/Rules/StarterWeb/AI/IndividualAuth/project.json:
```diff
@@ -53,8 +54,7 @@
       "imports": [
-        "portable-net45+win8+dnxcore50",
-        "portable-net45+win8"
+        "portable-net451+win8"
```
Are you sure this works? I've seen this pull in the wrong version of newtonsoft.net by not specifying +dnxcore50

natemcmaster replied:
In making this change, I am assuming ~~https://github.com/dotnet/cli/pull/3386~~ https://github.com/dotnet/cli/pull/3411 will be taken in which bumps to the netstandard version of JSON.NET

natemcmaster replied:
But also, "portable-net451+win8" works just fine in all of our testing.

phenning replied:
great!

phenning commented on src/Rules/StarterWeb/AI/IndividualAuth/project.json:
```diff
@@ -19,11 +19,15 @@
     },
+    "Microsoft.EntityFrameworkCore.SqlServer": "1.0.0-*",
+    "Microsoft.EntityFrameworkCore.SqlServer.Design": {
+      "version": "1.0.0-*",
```
Use `__AspNetCorePackageVersion__`here instead of 1.0.0-\* (only for the SqlServer.Design package, in all three project.json files

phenning commented:
One additional comment inline about using `__AspNetCorePackageVersion__`, looks good.

We should hold off merging until we get a coherence-signed build with the package rename.

Commit:
React to EF package renames and remove EF.Tools imports

natemcmaster commented:
@phenning I've updated with the changes and also removed imports from EF.Tools as they won't be necessary anymore.

:watch: will wait on a coherence-signed build with the changes from https://github.com/aspnet/EntityFramework/pull/5689
